NKL/Python.lean

```diff
@@ -124,10 +131,22 @@ structure Fun where
   body: List Stmt
   deriving Repr
 
+/-
+A kernel is collection of:
+  - the name of the main kernel function: `entry`
+  - functions, including the primary function and any functions
+    called by the primary func that we are able to parse
+  - arguments to the primary function, the positional arguments
+    are in the field `args` and the keyword argument are in the
+    field `kwargs`
+  - global variables referenced by any of the functions
+-/
 structure Kernel where
   entry : String
   funcs : List (String × Fun)
-  globals : List (String × Option String)
+  args : List Expr'
+  kwargs : List (String × Expr')
+  globals : List (String × Expr')
```

Comment:
Nice documentation string. What's an example of a global? Where does it come from?

Reply:
Will add a comment.